### DISCUSSION TITLE
Tensorflow Shape Error

### DISCUSSION THREAD

**Discussion by YulianaGomez** [deepmodeling/deepmd-kit#1399] [Q&A] [answered]
I'm currently running into a tensor shape error (full error attached):

ValueError: Cannot feed value of shape (3,) for Tensor 'd_sea_t_natoms:0', which has shape '(4,)'

My basic workflow is using data from the SLUSCHI database ( https://materialsdata.nist.gov/handle/11256/693 ) and using your dpdata.LabeledSystem conversion:

dsys = dpdata.LabeledSystem('/global/homes/y/yzamora/sluschi/OUTCAR')
dsys.to('deepmd/raw', 'deepmd_raw', set_size = dsys.get_nframes())

From here, I use your raw_to_set.sh to split the .raw files into 4 different training sets (train and validation). Then I changed the example water json file (attached). After that, I use dp train with this json file and get the ValueError from above and attached. I've tried tracing back where this error is coming from, but could not find where the shape is coming from. 

Any suggestions on how to fix this error? I've attached an example of one of the datasets from the splits, full error log, and input json file. 
[Error_input.zip](https://github.com/deepmodeling/deepmd-kit/files/7810341/Error_input.zip)
Or strings to pull
[data_0.zip](https://github.com/deepmodeling/deepmd-kit/files/7810327/data_0.zip)
?

**Comment by Yi-FanLi** [comment]
[Yuliana_run.tar.gz](https://github.com/deepmodeling/deepmd-kit/files/7810350/Yuliana_run.tar.gz)

Hi Yuliana,
Your problem might be caused by using the raw_to_set.sh script. This can be solved by correctly using dpdata.

I added a python script which uses dpdata to split the OUTCAR data into 4 datasets. It is poscartodata.py in the Yuliana/vasp directory in the tarball. The OUTCAR file was downloaded from the SLUSCHI database. It contains 80 frames, so I am splitting it into 4 sets, with 20 frames in each frame. You can change the values of Nsets and Neach to make it appropriate for other data. I also copied it into Yuliana/train and modified the input script a little bit to specify the data directory. You can run the training directly by `dp train input_sluschi.json` to test if it works. Moreover, I deleted the type_map.raw files in each dataset, since type_map has already been specified by the input script, and my program cannot read "Al" correctly but is reading "A" instead. 

You can find the necessary files in the attached file. I guess it should work for the training purpose. You can replace the training data with those you need. Don't hesitate to let me know if you have any other questions.

Best,
Yifan

**Reply by sinaDFT** [reply]
Hello @Yi-FanLi @YulianaGomez I actually have the same problem and get the same error. I realized that the .tar.gz is not available anymore. I am wondering if you can please share this with me or any other file that fixes this error? Thanks, Sina

**Comment by sinaDFT** [comment]
@Yi-FanLi @njzjz I have a question about the shape error happening when I want to train the DeepMD. I use dpdata to convert the OUTCAR data to the .npy files. After that I just simply split them to four datasets and when I try to train the model, I get the error 
"ValueError: Cannot feed value of shape (3,) for Tensor d_sea_t_natoms:0, which has shape (4,)". I am sure that the dimension of all tensors are correct. Could you please let me know what is the origin of this error and why I am getting this? Thanks.

**Comment by njzjz** [comment, marked as answer]
The above answer is incorrect. The reason of this error is that `type_map` has one type but `sel` has two elements.

**Reply by adidey1** [reply]
Can you please mention how to fix this? I have the same error and 1 atom type in type_map. I can not understand how to modify se1 and set it for 1 element. Is it included within the input.json file or somewhere else?

**Reply by njzjz** [reply]
It should be included in `input.json`. For details, check documentation:
- https://docs.deepmodeling.com/projects/deepmd/en/master/model/sel.html
- https://docs.deepmodeling.com/projects/deepmd/en/master/train/train-input.html
